src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -6547,20 +6549,30 @@ public override IOperation VisitDeclarationPattern(IDeclarationPatternOperation
                 operation.Syntax, IsImplicit(operation));
         }
 
-        public override IOperation VisitRecursivePattern(IRecursivePatternOperation operation, int? argument)
+        internal override IOperation VisitRecursivePattern(IRecursivePatternOperation operation, int? argument)
         {
             return new RecursivePatternOperation(
                 inputType: operation.InputType,
                 matchedType: operation.MatchedType,
                 operation.DeconstructSymbol,
                 operation.DeconstructionSubpatterns.SelectAsArray(p => (IPatternOperation)Visit(p)),
-                operation.PropertySubpatterns.SelectAsArray(p => (p.Item1, (IPatternOperation)Visit(p.Item2))),
+                operation.PropertySubpatterns.SelectAsArray(p => (IPropertySubpatternOperation)Visit(p)),
                 operation.DeclaredSymbol,
                 semanticModel: null,
                 operation.Syntax,
                 IsImplicit(operation));
         }
 
+        internal override IOperation VisitPropertySubpattern(IPropertySubpatternOperation operation, int? argument)
+        {
+            return new PropertySubpatternOperation(
+                semanticModel: null,
+                operation.Syntax,
+                IsImplicit(operation),
+                Visit(operation.Member),
+                (IPatternOperation)Visit(operation.Pattern));
```

Comment:
>(IPatternOperation)Visit(operation.Pattern) [](start = 16, length = 43)

It looks like in this case we are not setting the right substitution for the ```InstanceReferenceKind.PatternInput``` for the case when the ```operation.Pattern``` is a recursive pattern. Given that majority of nodes are internal, I'd suggest to not substitute ```InstanceReferenceKind.PatternInput``` with a capture in this PR and leave it as a regular node. We will have to figure out the right way to deal with this later.

Reply:
Added a note to [https://github.com/dotnet/roslyn/issues/33175](https://github.com/dotnet/roslyn/issues/33175) to follow up.

---
In reply to: [254440669](https://github.com/dotnet/roslyn/pull/33178#discussion_r254440669) [](ancestors = 254440669)